components/automate-gateway/handler/ssoconfig.go

```diff
@@ -18,13 +27,26 @@ type SsoConfig struct {
 	client         deployment.DeploymentClient
 }
 
+type PostConfig struct {
+	Ca_contents           string   `json:"ca_contents"  validate:"required"`
+	Sso_url               string   `json:"sso_url" validate:"required"`
+	Email_attr            string   `json:"email_attr"  validate:"required"`
+	Username_attr         string   `json:"username_attr" validate:"required"`
+	Groups_attr           string   `json:"groups_attr,omitempty"`
+	Allowed_groups        []string `json:"allowed_groups,omitempty"`
+	Entity_issuer         string   `json:"entity_issuer"  validate:"required"`
+	Name_id_policy_format string   `json:"name_id_policy_format,omitempty"`
+}
+
 // NewSsoConfigHandler - create a new ssoconfig service handler
 func NewSsoConfigHandler(license_client license_control.LicenseControlServiceClient, client deployment.DeploymentClient) *SsoConfig {
 	return &SsoConfig{
 		license_client: license_client,
 		client:         client,
 	}
 }
+var validate = validator.New()
```

Comment:
Please use this validator to validate the struct, right now its not used anywhere